File: source/verify/spi_verify.py

# spi_verify.py

import sys
import time

import read_spi_cmd
import read_buf_to_file

def spi_verify(ip_addr, port, file_size, conv_file_name) :
      
   file = open("convert_folder/verify_file.txt","wb")
   file.close()
   
   read_addr = 0
   
   while (read_addr < file_size) :
      
      read_spi_cmd.read_spi_cmd(ip_addr = ip_addr, port = port, spi_reg_addr = read_addr)
      
      time.sleep(0.005)
      
      read_buf_to_file.read_buf_to_file(ip_addr = ip_addr, port = port)
      
      read_addr = read_addr + 250
      
   
   conv_source_file = "convert_folder/spi_" + conv_file_name
   
   file1 = open(conv_source_file,"rb")
   file2 = open("convert_folder/verify_file.txt","rb")
   
   byte_counter = 0
   
   while 1 :
      comp1 = file1.read(1)
      comp2 = file2.read(1)
      
      if (comp1 == "") :
         print byte_counter
         break
      
      ord_comp1 = ord(comp1)
      ord_comp2 = ord(comp2)
      
      if ((ord_comp1 - ord_comp2) != 0) :
         print "Verify False"
         print byte_counter
         file1.close()
         file2.close()
         sys.exit(-1)
      
      byte_counter = byte_counter + 1
      
      if ((byte_counter % 10000) == 0) :
         print byte_counter
   
   file1.close()
   file2.close()
   
   print "Verify Success"
   


if __name__ == '__main__' :
   
   argvs = sys.argv
   argc  = len(argvs)
   
   if (argc != 2) :
      print "Usage : %s <source file>" % argvs[0]
      sys.exit(-1)
      
   
   spi_verify(ip_addr = '192.168.10.16', port = 4660, file_size = 341618 ,conv_file_name = argvs[1])

File: source/verify/spi_verify.cc

// spi_verify

#include "read_spi_cmd.h"
#include "read_buf_to_file.h"
#include "spi_verify.h"

using namespace std;

void spi_verify(const char *ip_addr, unsigned int port, unsigned int file_size, const char *conv_file_name)
{

  string verfile("convert_folder/verify_file_");
  verfile+=ip_addr; verfile+=".txt";
  ofstream file(verfile.c_str(),ios::out|ios::binary|ios::trunc);
  if(!file){
    cerr << "Cannot open "<< verfile << endl;
    return;
  }
  file.close();

  unsigned int read_addr = 0;
   
  while (read_addr < file_size) 
    {

      read_spi_cmd(ip_addr, port, read_addr);

      usleep(5000);

      read_buf_to_file(ip_addr, port,0x800,250,verfile.c_str());

      read_addr += 250;

    }

  string conv_source_file("convert_folder/spi_");
  conv_source_file+= conv_file_name;

  ifstream file1(conv_source_file.c_str(), ios::in | ios::binary);
  if(!file1){
    cerr << "Cannot open " << conv_source_file << endl;
    return;
  }
  ifstream file2(verfile.c_str(), ios::in | ios::binary);

  int byte_counter = 0;

  while(1)
    {
      char comp1,comp2;

      file1.read(&comp1,1);
      file2.read(&comp2,1);

      if (file1.gcount()==0){
	cout << byte_counter <<endl;
	break;
      }
      
      if ((comp1 - comp2) != 0){
	cerr << "Verify False" << endl;
	cerr << byte_counter << endl;
	file1.close();
	file2.close();
	exit(-1);
      }

      byte_counter++;

      if ((byte_counter % 10000) == 0)
	cout <<  byte_counter <<endl;
    }
  file1.close();
  file2.close();
   
  cout <<"Verify Success" <<endl;

}


#ifdef TEST_SPI_VERIFY
int main(int argc, char **argv)
{
  if (argc !=5){
    cout << "Usage : " << argv[0] << "<IP> <port> <size> <source file>" <<endl;
    exit(-1);
  }  

  spi_verify(argv[1], atoi(argv[2]), atoi(argv[3]) ,argv[4]);

  return 0;
}

#endif


